helm/templates/deployment.yaml

```diff
@@ -49,15 +49,24 @@ spec:
               port: http
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
+          envFrom:
+          {{- if not .Values.superblocks.existingSecret }}
+            - secretRef:
+                name: {{ include "superblocks-agent.fullname" . }}
+          {{- else }}
+            - secretRef:
+                name: {{ .Values.superblocks.existingSecret }}
```

Comment:
can we call it superblocks.agentKeySecretName

Reply:
addressed, calling it superblocks.agentKeyExistingSecret instead to not imply that this is the name we're creating the secret under